code/modules/mob/living/living.dm

```diff
@@ -2334,7 +2334,8 @@ GLOBAL_LIST_EMPTY(fire_appearances)
 		new_value = 0
 
 	. = usable_legs
-	usable_legs = new_value
+	if(!isnull(new_value))
+		usable_legs = new_value
```

Comment:
I think this is an inappropriate modification to this proc.  

Instead of allowing this to be called with `null` legs, have a separate proc `update_usable_leg_status` which handles adding/removing traits and updating the movespeed modifier. 

Then you can just call that proc from `set_usable_legs` and wherever you need to update it (cane equipping)